File: Codeforces/R727/new.py
# k = int(input())

# while k:
# 	n = int(input())
# 	x = int(input())
# 	t = int(input())
# 	count = 0
# 	z = n - int((t/x))
# 	y = int(n-z-1)
# 	count = z * int(t/x) + y*(y+1)/2
# 	print(int(count))

def main():
	k = int(input())	

	while k:
		n , x, t = [int(c) for c in input().split()]
		if n == 1:
			print(0)
			k = k-1
			continue
		b = int(t/x)
		# print(b, 'value')
		if b <= 0:
			print(0)
			k = k-1
			continue
		count = 0
		z = n - b
		y = (n-z-1)
		count = z * b + y*(y+1)/2
		print(int(count))
		k = k-1

if __name__ == "__main__":
   main()

File: Codeforces/R727/new.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long 
#define test(t) while(t--)
#define fo(i,a,n) for(int i = a; i < n; i++)
 
int main(){
#ifndef ONLINE_JUDGE
	freopen("input.txt","r", stdin);
	freopen("output.txt", "w", stdout);
#endif
	ll k, n,q;
	cin >> n >> q;
	string s;
	cin >> s;
	unordered_map<char, pair<ll, ll>>d;
	ll l,r;
	test(q){
	cin >> l >> r;
	// cout << l <<r;
	for(ll i = l-1; i < r; i++){
		// cout << s[i];
		d[s[i]].first++;
		d[s[i]].second = int(s[i])-int('a')+1;
	}
	ll count = 0;
	for(char i : s){
		if(d[i].first!=0){
			count += d[i].first*d[i].second;
			// cout << d[i].second << i<< " ";
			d[i].first = 0;
		}
	}
	cout << count <<endl;
	d.clear();

	}
	return 0;
}